applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c

```diff
@@ -187,51 +197,63 @@ int le_audio_send(uint8_t const *const data, size_t size)
 	static bool wrn_printed[CONFIG_BT_ISO_MAX_CHAN];
 	struct net_buf *buf;
 
+	/* TODO: This should all be based on number of audio channels */
+	uint8_t *p_data = (uint8_t *)data;
+	size_t num_streams = ARRAY_SIZE(streams);
+	size_t data_size = size/num_streams;
+
+
 	if (streams[0].ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
 		return -ECANCELED;
 	}
 
-	if (is_iso_buffer_full(0)) {
-		if (!wrn_printed[0]) {
-			LOG_WRN("HCI ISO TX overrun on ch %d - Single print", 0);
-			wrn_printed[0] = true;
+	/* for each stream send the associated coded data */
+	for (size_t i = 0U; i < num_streams; i++) {
+		if (is_iso_buffer_full(0)) {
+			if (!wrn_printed[i]) {
+				LOG_WRN("HCI ISO TX overrun on ch %d - Single print", i);
+				wrn_printed[i] = true;
+			}
+
+			return -ENOMEM;
 		}
 
-		return -ENOMEM;
-	}
+		wrn_printed[i] = false;
 
-	wrn_printed[0] = false;
+		buf = net_buf_alloc(iso_tx_pools[i], K_NO_WAIT);
+		if (buf == NULL) {
+			/* This should never occur because of the is_iso_buffer_full() check */
+			LOG_WRN("Out of TX buffers");
+			return -ENOMEM;
+		}
 
-	buf = net_buf_alloc(iso_tx_pools[0], K_NO_WAIT);
-	if (buf == NULL) {
-		/* This should never occur because of the is_iso_buffer_full() check */
-		LOG_WRN("Out of TX buffers");
-		return -ENOMEM;
-	}
+		net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+		net_buf_add_mem(buf, p_data, data_size);
 
-	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
-	net_buf_add_mem(buf, data, size);
+		/* Update pointer to next coded block */
+		p_data += data_size;
 
 #if (CONFIG_AUDIO_SOURCE_I2S)
-	struct bt_iso_tx_info tx_info = { 0 };
+		struct bt_iso_tx_info tx_info = { 0 };
 
-	ret = bt_iso_chan_get_tx_sync(streams[0].iso, &tx_info);
+		ret = bt_iso_chan_get_tx_sync(streams[i].iso, &tx_info);
 
-	if (ret) {
-		LOG_WRN("Error getting ISO TX anchor point: %d", ret);
-	} else {
-		audio_datapath_sdu_ref_update(tx_info.ts);
-	}
+		if (ret) {
+			LOG_WRN("Error getting ISO TX anchor point: %d", ret);
+		} else {
+			audio_datapath_sdu_ref_update(tx_info.ts);
```

Comment:
So we will always synchronise to stream index 0?

Reply:
Since this ts is based on the BIG anchor point (at the end of the interval), you should get the same ts for every stream. So doing it only for e.g. stream index 0 should be enough